adcircpy/forcing/winds/best_track.py

```diff
@@ -280,6 +319,18 @@ def make_request():
 
             self.__atcf = io.BytesIO(response.read())
 
+    @property
+    def track_length(self) -> float:
+        if not hasattr(self, '_track_length'):
+            lons, lats = self.df['longitude'], self.df['latitude']
+            distances = []
+            for index, _ in enumerate(lons):
```

Comment:
You can probably avoid the for loop if you replace math with numpy, but I leave this up to you to decide how to do it. If you're OK with the current implementation, I'm fine with it too. This is just a coding tip.

Reply:
okay yea makes sense to use vectorization as much as possible here if I have to process many tracks. I took a stab at this but need to verify that the distance calculation is correct.

I'm getting different track lengths than before with the non-vectorized calculation and the math lib functions.